apm-agent-tracer/src/main/java/co/elastic/apm/agent/tracer/ElasticContext.java

```diff
@@ -18,26 +18,113 @@
  */
 package co.elastic.apm.agent.tracer;
 
-public interface ElasticContext<T extends ElasticContext<T>> {
+import co.elastic.apm.agent.tracer.dispatch.BinaryHeaderSetter;
+import co.elastic.apm.agent.tracer.dispatch.TextHeaderGetter;
+import co.elastic.apm.agent.tracer.dispatch.TextHeaderSetter;
+import co.elastic.apm.agent.tracer.reference.ReferenceCounted;
+
+import javax.annotation.Nullable;
+
+public interface ElasticContext<T extends ElasticContext<T>> extends ReferenceCounted, Activateable<T> {
+
+    /**
+     * @return the span/transaction that is associated to this context, {@literal null} if there is none
+     */
+    @Nullable
+    AbstractSpan<?> getSpan();
```

Comment:
I would find `ElasticContext.getActive()` also confusing: Does it return the active span, baggage, or both?
What about `getAbstractSpan`? We already have the `AbstractSpan` class in the API indicating that it can be either a transaction or a span?

Reply:
Let's take `getAbstractSpan` if you think that's clearer, the return type also helps to understand it's definitely not a bagage here.